app/controllers/users_controller.rb

```diff
@@ -16,15 +25,52 @@ def create
     if @user.save
       log_in @user
       flash[:success] = t "controllers.users.welcome"
-      redirect_to @user
+      redirect_back_or user
     else
       flash[:danger] = t "controllers.users.acc_reg_fail"
       render :new
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @user.update_attributes user_params
+      flash[:success] = t "controllers.users.update_success"
+      redirect_to @user
+    else
+      flash[:danger] = t "controllers.users.update_fail"
+      render "edit"
```

Comment:
```suggestion
      render :edit
``` 
and add flash for this case

Reply:
done